### PR TITLE
fix(android): Reading unassigned background color property from Tab/TableViewRow crashes as of 8.1.0

### PR DESCRIPTION
**Pull request by jquick-axway**
**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-27231

**Summary:**
- This bug was never in a release version. We caught it before release.
- We were missing `null` checks when calling our Java `getOrCreateView()` method.

**Test:**
1. Build and run the below code on Android.
2. Verify that the app does not crash on startup. (This is the fix.)
3. Verify that the background colors for Tabs, TableView Section, and TableView Rows change from "red" to "orange" to "blue" and then back to "red".

```javascript
function updateColorsForView(view) {
	if (view) {
		switch (view.backgroundColor) {
			case "red":
				view.backgroundColor = "orange";
				break;
			case "orange":
				view.backgroundColor = "blue";
				break;
			default:
				view.backgroundColor = "red";
				break;
		}
		if (view.backgroundDisabledColor !== view.backgroundColor) {
			view.backgroundDisabledColor = view.backgroundColor;
		}
		if (view.backgroundFocusedColor !== view.backgroundColor) {
			view.backgroundFocusedColor = view.backgroundColor;
		}
		if (view.backgroundSelectedColor !== view.backgroundColor) {
			view.backgroundSelectedColor = view.backgroundColor;
		}
	}
}

function createTab(title) {
	var headerView = Ti.UI.createLabel({
		text: "TableView Section",
		textAlign: Ti.UI.TEXT_ALIGNMENT_CENTER,
		color: "white",
	});
	var section = Ti.UI.createTableViewSection({ headerView: headerView });
	var row1 = Ti.UI.createTableViewRow({ title: "Row 1" });
	var row2 = Ti.UI.createTableViewRow({ title: "Row 2" });
	section.add(row1);
	section.add(row2);
	
	var window = Ti.UI.createWindow({ title: title });
	window.add(Ti.UI.createTableView({
		data: [section],
		width: Ti.UI.FILL,
		height: Ti.UI.FILL,
	}));
	var tab = Ti.UI.createTab({
		title: title,
		window: window,
	});

	function updateAllViews() {
		updateColorsForView(headerView);
		updateColorsForView(section);
		updateColorsForView(row1);
		updateColorsForView(row2);
		updateColorsForView(tab);
	}
	updateAllViews();
	setInterval(function() {
		updateAllViews();
	}, 2000);

	return tab;
}

var tabGroup = Ti.UI.createTabGroup({
	tabs: [createTab("Tab 1"), createTab("Tab 2"), createTab("Tab 3")],
});
tabGroup.open();
```

### CODE REVIEW

File: android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java [hunk 1]
```diff
@@ -1029,24 +1030,33 @@ public TiViewProxy getParent()
...
-		if (getProperties().containsKeyAndNotNull(TiC.PROPERTY_BACKGROUND_COLOR)) {
-			return getProperties().getString(TiC.PROPERTY_BACKGROUND_COLOR);
+		// Return originally assigned property value, if available.
+		if (hasPropertyAndNotNull(TiC.PROPERTY_BACKGROUND_COLOR)) {
+			return TiConvert.toString(getProperty(TiC.PROPERTY_BACKGROUND_COLOR));
 		}
+
+		// Fetch the view.
+		TiUIView tiView = getOrCreateView();
+		if (view == null) {
```
Comment (ypbnv) [inline android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java]: I suppose the check here should be for `tiView`?

Comment (jquick-axway) [reply android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java]: Oh wow. Yes, good catch.

Member variable `view` happens to be the return value of `getOrCreateView()`, which is why this happened to work.

**Comment by jquick-axway** [comment]
Updated PR:
- Corrected `tiView` usage.
- Added `TableViewRow` unit tests.

**Review by garymathews — APPROVED** [review]
LGTM!

**Review by ssjsamir — APPROVED** [review]
FR Passed: Reading unassigned background color property from Tab/TableViewRow no longer crashes. Tested with the test steps mentioned aboive. 

**Test environment**
```
Pixel XL SIm (7.1.1)
JDK 1.8.0_131 
"NPM":"4.2.13","CLI":"7.1.0-master.24"}
Operating System Name: Mac OS Mojave
Operating System Version: 10.14.4
Node.js Version: 8.9.1
```

**Pull request merged**